src/lib/utils/dataUtils.ts

```diff
@@ -19,21 +19,109 @@ export function renderItems<T extends Element = HTMLElement>(
 ) {
   let generateStringTemplate;
   if (listElementWrapper) {
-    generateStringTemplate = () =>
-      data.reduce((html, d) => {
+    generateStringTemplate = () => {
+      const fragment = document.createDocumentFragment();
+      const div = document.createElement('div');
+      div.innerHTML = generateStringTemplate();
```

Comment:
Wouldn't this result in infinite recursion, calling itself over and over again? :)

Reply:
Ahh sorry, I'm not done yet, just pushed what I had at the end of the day